### PR TITLE
communites home screen context actions

### LINKED ISSUE #14327: Communities home screen context actions
Implement communities home screen bottom sheet as on the figma designs in the following link
https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?node-id=4444%3A193966

### PR DESCRIPTION
**Pull request by jo-mut**
fixes #14327

### Summary
This PR implement the communities home bottom sheet adds the functionality to `view members`, `invite people from contacts`, `mark as read` and `share community`.

### CODE REVIEW

File: src/status_im/ui2/screens/communities/communities_home.cljs [hunk 1]
```diff
@@ -28,12 +28,11 @@
...
-                     (>evt [:dismiss-keyboard])
-                     (>evt [:navigate-to :community {:community-id (:id community-item)}]))
-    :on-long-press #(>evt [:bottom-sheet/show-sheet
-                           {:content (fn []
-                                       [community/community-actions community-item])}])}
+                     #(re-frame/dispatch [:communities/load-category-states (:id community-item)])
+                     #(re-frame/dispatch [:dismiss-keyboard])
+                     #(re-frame/dispatch [:navigate-to :community {:community-id (:id community-item)}]))
+    :on-long-press #(re-frame/dispatch [:bottom-sheet/show-sheet
+                                        {:content (constantly [home-actions/communities-actions-sheet (:id community-item)])}])}
```
Comment (ibrkhalil) [inline src/status_im/ui2/screens/communities/communities_home.cljs]: Didn't we agree on using `rf/sub` and `rf/dispatch` ?

File: src/status_im/ui2/screens/communities/community_home_context_actions.cljs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+(ns status-im.ui2.screens.communities.community-home-context-actions
+  (:require [quo2.components.drawers.action-drawers :as drawers]
+            [status-im.i18n.i18n :as i18n]
+            [status-im.utils.re-frame :as rf]
+            [status-im.communities.core :as communities]))
+
+(defn hide-sheet-and-dispatch [event]
+  (rf/dispatch [:bottom-sheet/hide])
+  (rf/dispatch event))
```
Comment (ibrkhalil) [inline src/status_im/ui2/screens/communities/community_home_context_actions.cljs]: I believe this same logic exists in the bottom sheet namespace

File: src/status_im/ui2/screens/communities/community_home_context_actions.cljs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+              :accessibility-label :invite-people-from-contacts
+              :label               (i18n/label :t/invite-people-from-contacts)
+              :on-press            #(hide-sheet-and-dispatch [::communities/invite-people-pressed id])}
+             {:icon                :i/qr-code
+              :accessibility-label :show-qr
+              :label               (i18n/label :t/show-qr)}
+             {:icon                :i/share
+              :accessibility-label :share-community
+              :label               (i18n/label :t/share-community)
+              :on-press            #(hide-sheet-and-dispatch [::communities/share-community-pressed id])}]})
```
Comment (ibrkhalil) [inline src/status_im/ui2/screens/communities/community_home_context_actions.cljs]: Suggestion
Too much repetition here for my taste, Can be extracted into a function and create a vector of options, But that's up to you ..
For example
```clj
(defn home-action [{:keys [icon accessibility-label label on-press]}]
{:icon icon
:accessibility-label accessibility-label
:label label
:on-press on-press})

(defn home-actions [id opts]
{:actions (mapv home-action [-{}- -{}- -{}-]}
Where -{}- is the map of options
```

**Review by ibrkhalil — APPROVED** [review]
Non blocking comments

**Comment by jo-mut** [comment]
@J-Son89 I realized that communities home screen and community overview screen share the same context actions. I refactor your implementation of context actions on the overview screen so it can work for both screens. I would appreciate you comments on this.

**Comment by J-Son89** [comment]
> @J-Son89 I realized that communities home screen and community overview screen share the same context actions. I refactor your implementation of context actions on the overview screen so it can work for both screens. I would appreciate you comments on this.

nice one John! I took a look and all looks good to me :)

File: src/status_im2/contexts/communities/home/view.cljs [hunk 4]
```diff
@@ -12,9 +12,7 @@
             [utils.re-frame :as rf]
             [i18n.i18n :as i18n]
             [quo2.core :as quo]
-
-            ;; TODO move to status-im2
-            [status-im.ui.screens.communities.community :as community]))
+            [status-im.ui.screens.communities.community-options-bottom-sheet :as home-actions]))
```
Comment (flexsurfer) [inline src/status_im2/contexts/communities/home/view.cljs:15]: please add comments back, because its still should be implemented in 2 namespace

Comment (jo-mut) [reply src/status_im2/contexts/communities/home/view.cljs:15]: oh yeah

**Pull request merged**